src/app/app.component.ts

```diff
@@ -79,15 +129,32 @@ export class AppComponent implements AfterViewInit, OnDestroy {
     this.optionsVisible = !this.optionsVisible;
   }
 
-  toggleComponentsOverlay() {
-    this.componentsListVisible = !this.componentsListVisible;
+  public onFocus(): void {
+    this.isComponentListVisible = true;
   }
 
-  hideComponentsOverlay() {
-    this.componentsListVisible = false;
+  private handleArrowDown(): void {
+    if (
+      this.componentsListSearchService.activeElementIndex$.value + 1 >
+      this.componentsListSearchService.flatFilteredComponentLinkList$.value.length - 1
+    ) {
+      this.componentsListSearchService.activeElementIndex$.next(0);
+    } else {
+      this.componentsListSearchService.activeElementIndex$.next(
+        this.componentsListSearchService.activeElementIndex$.value + 1,
+      );
+    }
   }
 
-  showComponentsOverlay() {
-    this.componentsListVisible = true;
+  private handleArrowUp(): void {
+    if (this.componentsListSearchService.activeElementIndex$.value - 1 < 0) {
+      this.componentsListSearchService.activeElementIndex$.next(
+        this.componentsListSearchService.flatFilteredComponentLinkList$.value.length - 1,
+      );
+    } else {
+      this.componentsListSearchService.activeElementIndex$.next(
+        this.componentsListSearchService.activeElementIndex$.value - 1,
+      );
+    }
```

Comment:
Extract huge properties access lines into a variables with descriptive names, for example:
`this.componentsListSearchService.activeElementIndex$.value - 1` -> `prevElementIndex`
Intead of `if` use `this.componentsListSearchService.activeElementIndex$.next(Math.max(prevElementIndex, 0))`.